src/include/optimizer/stats/stats_storage.h

```diff
@@ -0,0 +1,127 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// stats_storage.h
+//
+// Identification: src/include/optimizer/stats/stats_storage.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "optimizer/stats/table_stats.h"
+#include "optimizer/stats/column_stats.h"
+
+#include <sstream>
+
+#include "common/macros.h"
+#include "type/types.h"
+
+#include "type/value_factory.h"
+
+namespace peloton {
+
+namespace storage {
+  class DataTable;
+  class Tuple;
+}
+
+namespace catalog {
+  class Schema;
+}
+
+namespace optimizer {
+
+#define STATS_TABLE_NAME "stats"
+
+#define SAMPLES_DB_NAME "samples_db"
+
+using ValueFrequencyPair = std::pair<type::Value, double>;
+
+class StatsStorage {
+ public:
+
+  // Global Singleton
+  static StatsStorage *GetInstance();
+
+  StatsStorage();
+
+  /* Functions for managing stats table and schema */
+
+  void CreateStatsCatalog();
+
+  std::unique_ptr<catalog::Schema> InitializeStatsSchema();
+
+  storage::DataTable *GetStatsTable();
+
+  /* Functions for adding, updating and quering stats */
+
+  void AddOrUpdateTableStats(storage::DataTable *table, TableStats *table_stats);
+
+  std::unique_ptr<storage::Tuple> GetColumnStatsTuple(
```

Comment:
It's for generating a tuple of column stats, which will be inserted into stats table. I think I can make this function private.